interview-questions.md

```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: A Ruby gem is a library of Ruby files which can be incorporated into a Ruby project in order to give the project additional functionality without having to build that functionality in from scratch.
 
-  Researched answer:
+  Researched answer: A gem is a third-party library which can be installed in Ruby in order to fill a specific need. Rails is an example of a popular Ruby gem.
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: Ruby on Rails is a back-end Ruby library which can be used to interface with databases and create web applications.
 
-  Researched answer:
+  Researched answer: Ruby on Rails is a server-side web application framework written in Ruby, which means that it is designed to access and interact with the data necessary to make a web application functional. For example, the HTML/CSS/Javascript that powers the front end of many websites does not have the ability to store data in any large quantities which means that things like Rails are necessary for access to customer login information, transaction details, data archival, etc.
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database is one that stores and organizes data depending on how the data relates to the other data in the database. A non-relational database would be one where information is stored without any sort of ordering or organizing system.
 
-  Researched answer:
+  Researched answer: I was mostly correct. A relational database stores data in tables and rows and makes the data relational by assigning each row of data a unique identifier (primary key). This key can be added to a record in another table where it is called the foreign key. This key connection allows data to be easily referenced between different tables and generally results in data accuracy and consistency. Non-relational databases stores data without the use of tables, rows, or keys. The method of data storage is typically optimatized for the requirements of the type of data being stored.
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: A primary key is the key in a database assigned to a piece of data which then makes that data easy to query. It's important because it provides a means of accessing that data.
 
-  Researched answer:
+  Researched answer: The primary key is a unique identifier assigned to each row of data in a relational database. This key can be added to a record in another table where it becomes a foreign key. The relationship between these keys allows data stored in different tables to be easily referenced and related to each other, which increases the accuracy and consistency of the queried data.
```

Comment:
Outstanding work in this section!  Your answers were descriptive and well thought-out, great work!